### DISCUSSION TITLE
Export?

### DISCUSSION THREAD

**Discussion by Pamasich** [aclist/kbin-megamod#215] [Active discussions]
Is there any way I could export the settings? I've set up the script on one device, but would like to use the same settings on another PC too, maybe my phone as well.

If there's no built in way yet (didn't find one so I assume that's the case), can I just copy over the ~~cookie(s)~~ (edit: I meant the local storage)? I see one that seems to contain all the settings. Will that be enough, or do they have to be manually turned on for the first time?

**Comment by aclist** [comment]
This is a good suggestion, thank you. There is currently no method actively implemented, but we could allow you to download a text file in order to restore the settings.

The local storage contains a list of the features that are enabled (controls the on/off status), consisting of a list of true/false values, and then the individual features themselves have a variety of sub-settings, stored under unique names for each group of settings. In the worst case scenario, if you are not able to manually splice this file in, if you at least take note of its contents, you'll know what your settings are.

If you can successfully copy that intact, it should work, but I can understand why this is not ideal. Given that there are 30+ discrete features within KES now, it seems reasonable to allow you to back up the granular settings and restore them. The thing we have to be careful about and design correctly here if adding this as a feature is parsing and validating the settings you paste back in, because if they contain any erroneous data/typos, it could lead to the settings being incorrectly set.

It's worth doing and I'll try to target the next release for adding that.

**Comment by aclist** [comment]
@Pamasich 

Updated backup functionality (export to file, import from file, and reset) is in the cooker and will be coming in the next release. It is now possible to save all current settings to a file and restore from that file again later. This functionality triggers a file dialog and lets the user select a text file they have backed up.

**Comment by aclist** [comment]
@Pamasich 

Hello, this feature has been added in KES 3.0.0. You can export/import to/from a file and it will restore all settings.
